packages/clients/meldemichel/src/store/module.ts

```diff
@@ -4,12 +4,21 @@
 import { PolarModule } from '@polar/lib-custom-types'
 
 interface SetMapStatePayload {
-  mapCenter: string // resembling 'number,number'
-  mapZoomLevel: number | string
-  mapBaseLayer: number | string
-  vendor_maps_position: string // resembling 'number,number'
-  vendor_maps_address_str: string
-  vendor_maps_address_hnr: number | string
+  mapCenter?: string // resembling 'number,number'
+  mapZoomLevel?: number | string
+  mapBaseLayer?: number | string
+  vendor_maps_position?: string // resembling 'number,number'
+  vendor_maps_address_str?: string
+  vendor_maps_address_hnr?: number | string
```

Comment:
The types don't correctly represent the actual values. All the values are always present in the object, but may return `undefined`, so an intersection with `undefined` would be the way to go

Reply:
The actual object referred to in `SetMapStatePayload` is not defined by us, but by the people installing our client. They may very well leave out any of the information provided.

The extending `GetMapState` indeed does always have all fields; however, directly typing `undefined` as any key's value further below (as done for `vendor_maps_address_hnr`) does not yield an error. Hence, TS accepts that `?` also allows for setting the value as `undefined`.

Reply:
I'd still press for typing the values differently but won't open a barrel if you won't 🎩